### PR TITLE
Add new parameter "functions" for chat completion API

### PR DESCRIPTION
**Pull request by samirkhoja**
Update chat completion API to include new parameter "functions" for model variant "0613" as described here: https://openai.com/blog/function-calling-and-other-api-updates

### CODE REVIEW

**Comment by samirkhoja** [comment]
Quickly did this in the GitHub UI so lint checks won't pass but just leaving it here as reference.

**Comment by coggsfl** [comment]
@samirkhoja thanks for posting - this is big news!  Azure version is only a few days away too.

File: chat.go [hunk 1]
```diff
@@ -26,23 +27,50 @@ type ChatCompletionMessage struct {
 	// the documentation for the official library for python:
```
Comment (j178) [inline chat.go:27]: `name` now is in the official documentation: https://platform.openai.com/docs/api-reference/chat/create#chat/create-name

File: chat.go [hunk 2]
```diff
@@ -26,23 +27,50 @@ type ChatCompletionMessage struct {
...
-	N                int                     `json:"n,omitempty"`
-	Stream           bool                    `json:"stream,omitempty"`
-	Stop             []string                `json:"stop,omitempty"`
-	PresencePenalty  float32                 `json:"presence_penalty,omitempty"`
-	FrequencyPenalty float32                 `json:"frequency_penalty,omitempty"`
-	LogitBias        map[string]int          `json:"logit_bias,omitempty"`
-	User             string                  `json:"user,omitempty"`
+	Model            string                   `json:"model"`
+	Messages         []ChatCompletionMessage  `json:"messages"`
+	Functions        []ChatCompletionFunction `json:"functions,omitempty"`
```
Comment (j178) [inline chat.go:63]: Another top-level parameter `function_call` should be added: https://platform.openai.com/docs/api-reference/chat/create#chat/create-function_call

**Pull request closed without merging**